### PR TITLE
Added batch document creation

### PR DESCRIPTION
**Pull request by cdaguerre**

### CODE REVIEW

File: lib/Textmaster/Api/AbstractApi.php [hunk 1]
```diff
@@ -106,9 +106,9 @@ public function getClient()
      *
      * @return \GuzzleHttp\Psr7\Stream|mixed|\Psr\Http\Message\StreamInterface
      */
-    protected function get($path, array $parameters = [], $requestHeaders = [])
+    protected function get($path, array $parameters = array(), $requestHeaders = array())
```
Comment (jmleroux) [inline lib/Textmaster/Api/AbstractApi.php]: Why do you prefer the old notation ?

File: lib/Textmaster/Model/Document.php [hunk 2]
```diff
@@ -389,12 +389,12 @@ protected function formatTranslatedContent()
      *
      * @param array $content
      */
-    private function checkArrayContent(array $content)
+    private function checkArrayContent(array &$content)
     {
-        foreach ($content as $value) {
-            if (!is_array($value) || !array_key_exists('original_phrase', $value)) {
+        foreach ($content as &$value) {
```
Comment (Pitoune) [inline lib/Textmaster/Model/Document.php]: no need for &

File: lib/Textmaster/Translator/Adapter/AbstractAdapter.php [hunk 3]
```diff
@@ -159,13 +162,27 @@ protected function getProperties($subject, array $properties, $language, $origin
...
             if ($original) {
                 $value = array('original_phrase' => $value);
             }
 
             $data[$property] = $value;
         }
 
+        if (!count($data)) {
+            throw new InvalidArgumentException(sprintf(
+                'Object of type "%s" has no translatable properties to translate (ie. non-empty). Checked for: "%s"',
```
Comment (Pitoune) [inline lib/Textmaster/Translator/Adapter/AbstractAdapter.php]: non-empty ?

File: lib/Textmaster/Model/Project.php [hunk 4]
```diff
@@ -219,7 +221,46 @@ public function createDocument()
...
+        $document = new Document($this->client, array('project_id' => $this->getId()));
+        $document->setProject($this);
+
+        return $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addDocuments(array $documents)
```
Comment (Pitoune) [inline lib/Textmaster/Model/Project.php]: why not making 2 private methods ? One to generate the data to send through the API. And another to dispatch events .

File: lib/Textmaster/Translator/Adapter/AbstractAdapter.php (hunk 3 above)
Comment (cdaguerre) [inline lib/Textmaster/Translator/Adapter/AbstractAdapter.php]: No non-empty values... Any other suggestion?

**Pull request merged**